File: src/main/scala/com/github/ai/split/entity/Group.scala
package com.github.ai.split.entity

import com.github.ai.split.utils.UuidUtils
import com.github.ai.split.utils.UuidUtils.EMPTY_UID

import java.util.UUID

case class Group(
  uid: UUID = EMPTY_UID,
  ownerUid: UUID,
  title: String,
  description: String,
  members: List[UUID],
  expenses: List[Expense]
)

File: android/app/src/main/java/com/github/ai/simplesplit/android/model/Group.kt
package com.github.ai.simplesplit.android.model

import com.github.ai.simplesplit.android.data.database.model.CurrencyEntity
import kotlinx.serialization.Serializable

@Serializable
data class Group(
    val uid: String,
    val title: String,
    val description: String,
    val currency: CurrencyEntity,
    val members: List<Member>,
    val expenses: List<Expense>,
    val paybackTransactions: List<Transaction>,
    val createdSeconds: Long,
    val modifiedSeconds: Long
)